galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/ITsqFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.cicsts;
+
+import javax.validation.constraints.NotNull;
+public interface ITsqFactory {
+
+      /**
+       * Create a new ITsq object with recoverable status   
+       * @param queueName TSQ name
+       * @param recoverable true for recoverable and false for non-recoverable
+       * @return ITsq object
+       * @throws TsqException if there is a problem in creating the ITsq object
+       */
+      public ITsq createQueue(@NotNull String queueName, @NotNull boolean recoverable) throws TsqException;
+    
+      /**
+       * Create a new ITsq object without recoverable status. Default recoverable status is NonRecoverable   
+       * @param queueName TSQ name
+       * @return ITsq object
+       * @throws TsqException if there is a problem in creating the ITsq object
```

Comment:
Extra comment perhaps on the @return ?
Something like "The existence of the ITsq object does not have any correlation to whether the queue actually exists underneath it. It is used to access or create the ITsq.

Or perhaps we need something like that on the ITsq interface.

Reply:
Updated the comment to add "The existence of the ITsq object does not have any correlation to whether the queue actually exists underneath it. It is used to access or create the ITsq." in @return.